### PR TITLE
[cleanup] getType should have explicit index.

### PR DESCRIPTION
**Pull request by rdzhabarov**
There were several bugs exposed by missing index.
This PR changes the behavior to use an explicit index.

### CODE REVIEW

**Comment by jfix71** [comment]
Thanks for working on this!

It seems most if not all the places you have to provide index `0` are with Variables. I'm wondering if it might make sense to add a `getType()` in `Variable` that calls `Node::getType(0)`, since Variables can only have one result.

**Comment by opti-mix** [comment]
In general, if we were calling/still want to call getType() without an index, we should probably assert if the node has more than one result.

**Comment by nadavrot** [comment]
@opti-mix We had an assertion that checked that there is only one result if we use the default argument version.

**Comment by rdzhabarov** [comment]
"I'm wondering if it might make sense to add a getType() in Variable that calls Node::getType(0), since Variables can only have one result."
@jfix71 right, that was for vars.
I think it's a good suggestion. Let me see what I can do here.

**Comment by rdzhabarov** [comment]
Majority of the getType(0) is gone now.

**Comment by rdzhabarov** [comment]
squashed second commit.

**Comment by qcolombet** [comment]
> "I'm wondering if it might make sense to add a getType() in Variable that calls Node::getType(0), since Variables can only have one result."
> @jfix71 right, that was for vars.
> I think it's a good suggestion. Let me see what I can do here.

Actually, now that I am replacing the direct uses of Variable with Load, I have more place to change. I could play the same trick though (have a getType() on the Load node).

File: include/glow/Graph/Node.h [hunk 1]
```diff
@@ -231,7 +231,7 @@ class Node : public Named,
   ~Node() { releaseUsers(); }
 
   /// \returns the n'th result type of the node.
-  TypeRef getType(unsigned idx = -1) const;
+  TypeRef getType(unsigned idx) const;
```
Comment (qcolombet) [inline include/glow/Graph/Node.h]: Node::getType has a special case for idx == -1.
This is not used anymore, right?
In other words, we should remove that code (I don't think this PR does that).

**Pull request merged**